erpnext/patches/v5_4/cleanup_journal_entry.py

```diff
@@ -0,0 +1,14 @@
+import frappe
+
+def execute():
+	for doctype, fieldname in (
+		("Sales Invoice", "against_invoice"),
+		("Purchase Invoice", "against_voucher"),
+		("Sales Order", "against_sales_order"),
```

Comment:
Change the order of "Sales Order" and "Sales Invoice", so that if both exists, sales invoice gets priority
Same for purchase